lib/DxilCompression/miniz.c

```diff
@@ -1147,7 +1149,7 @@ static mz_bool tdefl_compress_lz_codes(tdefl_compressor *d)
         if (pOutput_buf >= d->m_pOutput_buf_end)
             return MZ_FALSE;
 
-        *(mz_uint64 *)pOutput_buf = bit_buffer;
+        memcpy((void*)pOutput_buf, (void*)&bit_buffer, sizeof(bit_buffer));
```

Comment:
Similar question here to above.  Is memcpy used to obfuscate so UBSan doesn't complain?  Is there another way?

Reply:
Same answer as above. This doesn't obfuscate, it fixes the unaligned data reads.